JumpscaleLibs/sal/reservation_chatflow/reservation_chatflow.py

```diff
@@ -921,10 +922,17 @@ def delegate_domains_list(self, customer_tid, currency=None):
                 domains[dom.domain] = dom
         return domains
 
+    @staticmethod
+    def node_is_up(last_updated_time):
+        update_range = datetime.datetime.utcfromtimestamp(last_updated_time) + datetime.timedelta(minutes=10)
+        return update_range >= datetime.datetime.utcnow()
+
     def gateway_list(self, bot, currency=None):
         unknowns = ["", None, "Uknown", "Unknown"]
         gateways = {}
         for g in j.sal.zosv2._explorer.gateway.list():
+            if not self.node_is_up(g.updated):
```

Comment:
https://github.com/threefoldtech/jumpscaleX_libs/blob/master/JumpscaleLibs/sal/zosv2/node_finder.py#L13

why dont you use existing ?